examples/speculative_decoding/distributed_generate/worker.sh

```
@@ -14,52 +14,198 @@
+if [ "$SGLANG_TP_SIZE" -eq 1 ]; then
+    for gpu in $(seq 0 $((NUM_TEMPERATURES - 1))); do
+        port=$((BASE_PORT + gpu))
+        SERVER_PORTS+=("$port")
+        if [ "$BACKEND" = "vllm" ]; then
+            CUDA_VISIBLE_DEVICES=$gpu vllm serve /model/ \
+                --tensor-parallel-size 1 --served-model-name "$MODEL_NAME" \
+                --port "$port" --host 0.0.0.0 --trust-remote-code &
+        else
+            CUDA_VISIBLE_DEVICES=$gpu python3 -m sglang.launch_server \
+                --model-path /model --served-model-name "$MODEL_NAME" --tp 1 \
+                --port "$port" --host 0.0.0.0 ${SGLANG_EXTRA_ARGS:-} --trust-remote-code &
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)